cdp4jsonserializer/src/test/java/cdp4jsonserializer/Cdp4JsonSerializerImplTest.java

```diff
@@ -255,4 +282,110 @@ boolean areEqualBySumOfBytes(String s1, String s2) {
     System.out.println(s1SumOfBytes + "--" + s2SumOfBytes);
     return s1SumOfBytes == s2SumOfBytes;
   }
+
+  private static String getJsonString() {
+    return String.join("\\n",
+        "{\"widget\": {",
+        "\"debug\": \"on\",",
+        "\"window\": {",
+        "\"title\": \"Sample Konfabulator Widget\",",
+        "\"name\": \"main_window\",",
+        "\"width\": 500,",
+        "\"height\": 500",
+        "},",
+        "\"image\": {",
+        "\"src\": \"Images/Sun.png\",",
+        "\"name\": \"sun1\",",
+        "\"hOffset\": 250,",
+        "\"vOffset\": 250,",
+        "\"alignment\": \"center\"",
+        "},",
+        "\"text\": {",
+        "\"data\": \"Click Here\",",
+        "\"size\": 36,",
+        "\"style\": \"bold\",",
+        "\"name\": \"text1\",",
+        "\"hOffset\": 250,",
+        "\"vOffset\": 100,",
+        "\"alignment\": \"center\",",
+        "\"onMouseUp\": \"sun1.opacity = (sun1.opacity / 100) * 90;\"",
+        "}",
+        "}",
+        "}");
+  }
+
+  private static String getXmlString() {
+
+    return String.join("\\n",
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?>",
+        "<bookstore>",
+        "<book category=\"cooking\">",
+        "<title lang=\"en\">Everyday Italian</title>",
+        "<author>Giada De Laurentiis</author>",
+        "<year>2005</year>",
+        "<price>30.00</price>",
+        "<data><![CDATA[Within this Character Data block I can",
+        "use double dashes as much as I want (along with <, &, ', and \")",
+        "*and* %MyParamEntity; will be expanded to the text",
+        "\"Has been expanded\" ... however, I can't use",
+        "the CEND sequence.If I need to use CEND I must escape one of the",
+        "brackets or the greater-than sign using concatenated CDATA sections.",
+        "\"]]></data>",
+        "</book>",
+        "<book category=\"children\">",
+        "<title lang=\"en\">Harry Potter</title>",
+        "<author>J K. Rowling</author>",
+        "<year>2005</year>",
+        "<price>29.99</price>",
+        "</book>",
+        "<book category=\"web\">",
+        "<title lang=\"en\">Learning XML</title>",
+        "<author>Erik T. Ray</author>",
+        "<year>2003</year>",
+        "<price>39.95</price>",
+        "</book>",
+        "</bookstore>");
+  }
+
+  private static String[] getTestStrings() {
+    String[] strings = {
+        "value with trailing spaces  ",
+        "value with trailing space ",
+        " value with leading spaces",
+        "  value with leading space",
+        "=2*(2+2)",
+        "=2*\\n(2+2)",
```

Comment:
According to this website (https://www.baeldung.com/java-string-newline), a newline character in java is \n.
I think \\\r is something else: a "real" backslash followed by the letter 'r'.
I don't know if Java knows both \\\n and \\\r, but you get the general idea I presume.

This might be the reason why you haven't noticed that the first mentioned comment about the SingleLine / DOTALL flag has to be added.
I think there isn't a linebreak character in the string at this moment because of the double backslash, so the SingleLine/dotall flag is not necessary for the test to complete succesfully.
I expect the tests to fail when you use single backslashes here and the test will complete succesfully again when the dotall flag is added in de Pattern.Compile method.